### DISCUSSION TITLE
tokio::sync::RwLock has no support for serde Serialize and Deserialize?

### DISCUSSION THREAD

**Discussion by CSRaghunandan** [tokio-rs/tokio#4380] [Q&A] [answered]
I am trying to convert a piece of code that used std::sync::RwLock to tokio::sync::RwLock, but i'm getting the below errors:

```
 the trait `_::_serde::Serialize` is not implemented for `tokio::sync::RwLock<MyType>
```

**Comment by Darksonn** [comment, marked as answer]
It's not possible because reading the data in the `RwLock` requires taking a read lock, but serde is not performing the serialization in an async environment, so it cannot do the `.await` necessary to lock the rw-lock.

**Comment by CSRaghunandan** [comment]
Is there an async version of serde which can serialize and deserialize tokio's RwLock?

**Reply by Darksonn** [reply]
No.
